### PR TITLE
Translate RVSDG loops to tree encoding 

### PR DESCRIPTION
**Pull request by oflatt**
Extends the translation from RVSDG to translate loops, and adds two tests.
Makes some convenient macros for constructing asts.

### CODE REVIEW

File: src/rvsdg/tree_unique/to_tree.rs [hunk 1]
```diff
@@ -55,37 +61,39 @@ impl<'a> RegionTranslator<'a> {
...
-        for binding in translator.bindings.into_iter().rev() {
-            expr = cbind(binding, expr);
+        for binding in self.bindings.iter().rev() {
+            expr = cbind(binding.clone(), expr);
         }
         expr
     }
 
-    fn translate_operand(&mut self, operand: Operand) -> usize {
+    /// Returns a pure expression (e.g. `getarg(0)`) that
```
Comment (ajpal) [inline src/rvsdg/tree_unique/to_tree.rs:86]: I don't understand the e.g. What is getarg?

File: src/rvsdg/tree_unique/to_tree.rs [hunk 2]
```diff
@@ -55,37 +61,39 @@ impl<'a> RegionTranslator<'a> {
...
+    fn translate_operand(&mut self, operand: Operand) -> Expr {
         match operand {
-            Operand::Arg(index) => index,
-            Operand::Id(id) => self.translate_node(id),
-            Operand::Project(_id, _indexx) => {
-                todo!("Doesn't handle subregions yet");
+            Operand::Arg(index) => getarg(index),
+            Operand::Id(id) => getarg(self.translate_node(id)),
+            Operand::Project(p_index, id) => {
+                let index = self.translate_node(id);
```
Comment (ajpal) [inline src/rvsdg/tree_unique/to_tree.rs:95]: Why is this called index? Isn't this a region?

File: src/rvsdg/tree_unique/to_tree.rs [hunk 3]
```diff
@@ -135,8 +166,9 @@ impl<'a> RegionTranslator<'a> {
             BasicExpr::Print(args) => {
                 assert!(args.len() == 2, "print should have 2 arguments");
                 let arg1 = self.translate_operand(args[0]);
+                // should be unit
```
Comment (ajpal) [inline src/rvsdg/tree_unique/to_tree.rs:169]: What should be unit?

**Review by ajpal — APPROVED** [review]
couple questions, but lgtm!

File: src/rvsdg/tree_unique/to_tree.rs (hunk 1 above)
Comment (oflatt) [reply src/rvsdg/tree_unique/to_tree.rs:86]: `getarg` is a helper that desugars to `(Get (Arg (Id 0)) 0)`

File: src/rvsdg/tree_unique/to_tree.rs (hunk 2 above)
Comment (oflatt) [reply src/rvsdg/tree_unique/to_tree.rs:95]: The whole region becomes a let-bound tuple, so this index is the index of that tuple in the environment 
Added some comments that explain this better

File: src/rvsdg/tree_unique/to_tree.rs (hunk 3 above)
Comment (oflatt) [reply src/rvsdg/tree_unique/to_tree.rs:169]: Added a comment- argument 2 is the incoming print edge state, which we translate to unit in the tree encoding.

**Pull request merged**